core/ui/EditorToolbar/picture-dropdown.tid

```diff
@@ -4,8 +4,44 @@ title: $:/core/ui/EditorToolbar/picture-dropdown
 [img[$(imageTitle)$]]
 \end
 
+\define insert-external-image()
```

Comment:
> Ideally nothing would show up initially until the user typed something, since if there are lots of internal images, loading them all for displaying is quite slow and locks up the browser.

I think rather than not display anything, it would be better to limit the results, e.g. to 50 or 100.

> For external images tiddlywiki would try to display a small preview/thumbnail in the dropdown of the resulting path (as you suggested) confirming it was valid

Cool idea and possibly rather simple: just preview the output as it would look.